src/lib/NodesIntersectionObserver.test.ts

```diff
@@ -0,0 +1,193 @@
+import { Mock } from 'vitest';
+
+import { mockBoundingClientRect } from '../__tests__/utils';
+import { NodesIntersectionObserver } from './NodesIntersectionObserver';
+
+require('intersection-observer');
+
+const resetElementPosition = (
+	node: HTMLElement,
+	{
+		width = 100,
+		height = 100,
+		x = 0,
+		y = 0,
+	}: {
+		width?: number;
+		height?: number;
+		x?: number;
+		y?: number;
+	} = {},
+) => {
+	mockBoundingClientRect(node, {
+		width,
+		height,
+		x,
+		y,
+	});
+	// simulate a scroll event; the polyfill listens for the "scroll" event on the document
+	// The polyfill starts recalculating element positions only after the event
+	document.dispatchEvent(new Event('scroll'));
+};
+
+const waitMockCall = (callback: Mock, timeout = 200) => {
```

Comment:
Let's rename to `waitForMockCall`.

Also it would be nice to return an array of calls, to let check it immediately, instead of write two expects

Reply:
And let's move it to utils